cli/src/main/kotlin/commands/UploadResultToSw360Command.kt

```diff
@@ -89,9 +107,27 @@ class UploadResultToSw360Command : CliktCommand(
         getProjectWithPackages(ortResult).forEach { (project, pkgList) ->
             val linkedReleases = pkgList.mapNotNull { pkg ->
                 val name = createReleaseName(pkg.id)
-                sw360ReleaseClient.getSparseReleaseByNameAndVersion(name, pkg.id.version)
+                var release = sw360ReleaseClient.getSparseReleaseByNameAndVersion(name, pkg.id.version)
                     .flatMap { sw360ReleaseClient.enrichSparseRelease(it) }
                     .orElseGet { createSw360Release(pkg, sw360ReleaseClient) }
+
+                if (attachSources) {
+                    val tempDirectory = createTempDirectory(pkg.id.toCoordinates()).toFile()
+                    val saveSourcesTemporarily = saveSourcesTemporarily(pkg, tempDirectory)
+
+                    saveSourcesTemporarily?.let {
+                        release = uploadAttachmentToRelease(
+                            sw360ReleaseClient,
+                            release,
+                            saveSourcesTemporarily.toPath(),
+                            SW360AttachmentType.SOURCE
+                        )
+                    }
+
+                    tempDirectory.safeDeleteRecursively(force = true)
```

Comment:
To make sure that the directory is always deleted - even when exceptions occur -, the whole block should probably be wrapped in a try-finally statement.